test/metabase/query_processor/pivot_test.clj

```diff
@@ -254,6 +254,68 @@
                      :row_count 156}
                     (qp.pivot/run-pivot-query (assoc query :info {:visualization-settings viz-settings}))))))))))
 
+(deftest ^:parallel nested-question-pivot-aggregation-names-test
+  (testing "#43993"
+    (mt/dataset
+      test-data
+      (testing "Column aliasing needs to work even with joins and across stages"
+        (let [query        (mt/mbql-query
+                             orders {:source-query {:source-table $$orders
+                                                    :joins        [{:source-table $$people
+                                                                    :alias        "People - User"
+                                                                    :condition
+                                                                    [:= $orders.user_id
+                                                                     [:field %people.id {:join-alias "People - User"}]]}]
+                                                    :aggregation  [[:sum $subtotal]]
+                                                    :breakout     [!month.created_at
+                                                                   [:field %people.id {:join-alias "People - User"}]]}
+                                     :aggregation  [[:sum [:field "sum" {:base-type :type/Number}]]]
+                                     :breakout     [[:field "ID" {:base-type :type/Number}]]})
+              viz-settings (mt/$ids orders {:pivot_table.column_split
+                                            {:columns     [[:field "ID" {:base-type :type/Number}]]}})]
+          (testing "for a regular query"
+            (is (=? {:status :completed}
+                    (qp/process-query query))))
+          (testing "and a pivot query"
+            (is (=? {:status    :completed
+                     :row_count 1747}
+                    (-> query
+                        (assoc :info {:visualization-settings viz-settings})
+                        qp.pivot/run-pivot-query)))))))))
+
+(deftest model-with-aggregations-nested-pivot-aggregation-names-test
+  (testing "#43993"
+    (mt/dataset
+      test-data
```

Comment:
is it necessary to use mt/dataset for `test-data`?

Reply:
I'm not sure. I think if I want to mutate the DB I do need to set it up like this? I'm not very familiar, I was just following how it's done in similar mutation-y tests in this namespace.

I'll experiment with it.

Reply:
Cleaned this up, and it still seems to work. I think `mt/with-temp` is sufficient for this one.